nuxeo-core/nuxeo-core-convert/src/main/java/org/nuxeo/ecm/core/convert/extension/Converter.java

```diff
@@ -50,4 +50,18 @@
      */
     BlobHolder convert(BlobHolder blobHolder, Map<String, Serializable> parameters) throws ConversionException;
 
+    /**
+     * To override with converter specific logic only if needed.
+     * <br>
+     * Secondary method to handle the real Conversion Job.
+     * <p>
+     * Returned {@link BlobHolder} must implement {@link CachableBlobHolder}, otherwise result won't be cached.
+     * @param destinationMimeType to compare with inputMimeType
+     * @return a BlobHolder specific to the converter override or the result of {@link CommandLineBasedConverter#convert(BlobHolder, Map)}
+     * @since 11.1
+     */
+    default BlobHolder convert(BlobHolder blobHolder, Map<String, Serializable> parameters, String destinationMimeType)
```

Comment:
Converters should be aware of their `destinationMimeType` during creation/initialisation and decide to use these information if needed. Here we can call a converter converting to `text/html` with the `image/png` mimeType which makes non sense.
Furthermore, improving our converters initialisation phase could help further improvements. For instance, we can have a new parameter in the descriptor such as `idempotent` which tells the generic implementation to return the input blob if it has the same mimeType than the desired one.

Reply:
Done ! The parameter is now named bypassIfSameMimeType as discussed with @troger